src/actions.py

```diff
@@ -76,37 +76,41 @@ def add_equation_action(_action, _target, self):
 
 
 def select_all_action(_action, _target, self):
-    for item in self.get_data():
+    data = self.get_data()
+    for item in data:
         item.set_selected(True)
-    self.get_clipboard().add()
+    data.add_history_state()
 
 
 def select_none_action(_action, _target, self):
-    for item in self.get_data():
+    data = self.get_data()
+    for item in data:
         item.set_selected(False)
-    self.get_clipboard().add()
+    data.add_history_state()
 
 
 def undo_action(_action, _target, self):
-    self.get_clipboard().undo()
+    self.get_data().undo()
 
 
 def redo_action(_action, _target, self):
-    self.get_clipboard().redo()
+    self.get_data().redo()
 
 
 def optimize_limits_action(_action, _target, self):
-    utilities.optimize_limits(self)
+    self.get_data().optimize_limits(self)
 
 
 def view_back_action(_action, _target, self):
-    if self.get_window().get_view_back_button().get_sensitive():
-        self.get_view_clipboard().undo()
+    data = self.get_data()
+    if data.props.view_back_possible:
```

Comment:
Just a suggestion, but perhaps `data.props.can_view_back` would make sense? It's a bit shorter, and has the same meaning in the end.